helm-charts/splunk-otel-collector/templates/network-explorer/k8s-collector-deployment.yaml

```diff
@@ -0,0 +1,162 @@
+{{- if and .Values.networkExplorer.enabled .Values.networkExplorer.k8sCollector.enabled }}
+{{ $gateway := fromYaml (include "splunk-otel-collector.gateway" .) }}
+# The k8s-collector consists of two services:
+# 1) k8s-watcher: talks to the Kubernetes API server to determine the current state of
+#    the cluster; sets up watches to be notified of subsequent changes to pods, services
+#    and other resources.
+# 2) k8s-relay: relays the information collected by k8s-watcher to the reducer.
```

Comment:
Looks like this functionality supported by [collector k8sattributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor). Not asking for any changes in scope of this PR, just thoughts for future integration: I believe we can adjust the k8s attribute processor for this use case and use it instead of this component given that we are sending data through the collector anyway.